include/oboe/AudioStreamBuilder.h

```diff
@@ -438,28 +438,83 @@ class AudioStreamBuilder : public AudioStreamBase {
      */
-    AudioStreamBuilder *setDataCallback(std::shared_ptr<AudioStreamDataCallback> sharedDataCallback) {
+    AudioStreamBuilder *setDataCallback(
+            std::shared_ptr<AudioStreamDataCallback> sharedDataCallback) {
         // Use this raw pointer in the rest of the code to retain backwards compatibility.
         mDataCallback = sharedDataCallback.get();
         // Hold a shared_ptr to protect the raw pointer for the lifetime of the stream.
         mSharedDataCallback = sharedDataCallback;
+        mSharedPartialDataCallback.reset();
+        mPartialDataCallback = nullptr;
         return this;
     }
 
     /**
     * Pass a raw pointer to a data callback. This is not recommended because the dataCallback
     * object might get deleted by the app while it is being used.
     *
+    * If both this method and setPartialDataCallback(std::shared_ptr<AudioStreamPartialDataCallback>)
+    * are called, the data callback from the last called method will be used.
+    *
     * @deprecated Call setDataCallback(std::shared_ptr<AudioStreamDataCallback>) instead.
     * @param dataCallback
     * @return pointer to the builder so calls can be chained
     */
     AudioStreamBuilder *setDataCallback(AudioStreamDataCallback *dataCallback) {
         mDataCallback = dataCallback;
         mSharedDataCallback = nullptr;
+        mPartialDataCallback = nullptr;
+        mSharedPartialDataCallback.reset();
+        return this;
+    }
+
+    /**
+     * Specifies an object to handle data related callbacks from the underlying API.
+     *
+     * <strong>Important: See AudioStreamPartialDataCallback for restrictions on what may be called
+     * from the callback methods.</strong>
+     *
+     * We pass a shared_ptr and cache it so that the partial data callback object cannot be deleted
+     * before the stream is deleted.
+     *
+     * If both this method and setDataCallback(AudioStreamDataCallback*) or
+     * setDataCallback(std::shared_ptr<AudioStreamDataCallback>) are called,
+     * the data callback from the last called method will be used.
+     *
+     * Note that partial data callback from aaudio API at API level 37. In that case, when partial
+     * data callback is set on the Android device that is not supporting partial data callback API,
+     * the stream will fail to open.
+     *
+     * Also note that partial data callback is only supported by aaudio API. OpenSLES has been
+     * deprecated for years. When setting parital data callback and using openSLES will result in
+     * failing to open.
+     *
+     * When the stream is in low latency mode, the data buffer is pretty small. In that case, it
+     * may be easier to use AudioStreamDataCallback instead of AudioStreamPartialDataCallback. For
+     * other use cases that use a big buffer, such as offload playback, deep buffer playback, it
+     * will make more sense to use partial data callback. When the stream is offloaded, no data
```

Comment:
What's the quantifiable benefit of a partial data callback? Can we add comments to the normal data callback to use the partial callback in certain cases. Perhaps the "This is not suitable for compressed audio data, for gapless audio playback, or to drain audio to a fixed arbitrary stop point in frames." language in b/406332239 is a good start

Reply:
Thanks for the suggestion! I updated the comment.